iris_grib/_save_rules.py

```diff
@@ -576,9 +591,7 @@ def grid_definition_section(cube, grib):
 
     if isinstance(cs, GeogCS):
         if not regular_x_and_y:
-            raise iris.exceptions.TranslationError(
-                'Saving an irregular latlon grid to GRIB (PDT3.4) is not '
-                'yet supported.')
+            grid_definition_template_4(cube, grib)
```

Comment:
Surely there should be an "else" here ??
Otherwise it calls `grid_definition_template_4` **and then** `grid_definition_template_0`
Which sounds wrong ...

Reply:
Yup, I think there should 😳 Fixing this should get that test passing too 👍